profiling/src/lib.rs

```diff
@@ -268,45 +271,31 @@ extern "C" fn prshutdown() -> ZendResult {
     ZendResult::Success
 }
 
-/// # Safety
-/// Only call during an active request from the PHP thread!
-unsafe fn getenv(name: &CStr) -> Option<String> {
-    // CStr doesn't have a len() so turn it into a slice.
-    let name = name.to_bytes();
-
-    // Safety: called CStr, so invariants have all been checked by this point.
-    let val = sapi_getenv(name.as_ptr() as *const c_char, name.len());
-    let val = val.into_string();
-    if !val.is_empty() {
-        return Some(val);
-    }
-
-    /* If the sapi didn't have an env var, try the libc.
-     * Safety: pointer comes from valid CStr.
-     */
-    let val = libc::getenv(name.as_ptr() as *const c_char);
-    if val.is_null() {
-        return None;
-    }
-
-    /* Safety: `val` has been checked for NULL, though I haven't checked that
-     * `libc::getenv` always return a string less than `isize::MAX`.
+fn parse_boolean(string: &str) -> Option<bool> {
+    /* Care was taken to avoid allocating and to not do more than 2 calls to
+     * `eq_ignore_ascii_case` as this function gets called 2+ times on every
+     * single request that comes in. Not particularly performance sensitive,
+     * but it adds directly to latency of the end-user, and we also want to
+     * avoid allocator churn when we can.
      */
-    let val = CStr::from_ptr(val);
-    return Some(String::from_utf8_lossy(val.to_bytes()).into_owned());
-}
 
-fn parse_boolean(mut string: String) -> Option<bool> {
-    // no boolean strings are longer than 5 characters
-    if string.is_empty() || string.len() > 5 {
+    let n_bytes = string.len();
+    // no boolean strings are longer than 5 ASCII characters
+    if n_bytes == 0 || n_bytes > 5 {
         return None;
     }
 
-    string.make_ascii_lowercase();
-    match string.as_str() {
-        "1" | "on" | "yes" | "true" => Some(true),
-        "0" | "no" | "off" | "false" => Some(false),
-        _ => None,
+    // The empty string will not match since it was handled above.
```

Comment:
Any reason the empty string is included in the set if it never matches?

Reply:
Yes -- the lookup table is structured based on string length. There needs to be something there to compare to, it's just there are no truthy boolean strings that are 5 characters, and no falsey boolean strings that are 4 characters.

Reply:
Wow, is this something obvious or should it warrant a comment?

Reply:
Ah I see, it's an index into the table. I see the optimization :-) The comment was confusing me somehow.